apacheconfig/wloader.py

```diff
@@ -0,0 +1,160 @@
+#
+# This file is part of apacheconfig software.
+#
+# Copyright (c) 2018-2019, Ilya Etingof <etingof@gmail.com>
+# License: https://github.com/etingof/apacheconfig/LICENSE.rst
+#
+
+
+from apacheconfig import make_parser
+from apacheconfig import make_lexer
+
+import abc
+import six
+
+
+def _restore_original(word):
+    """If the `word` is a Quoted string, restores it to original.
+    """
+    if getattr(word, 'is_single_quoted', False):
+        return "'%s'" % word
+    if getattr(word, 'is_double_quoted', False):
+        return '"%s"' % word
+    return word
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Node():
+    """Generic class containing data that represents a node in the config AST.
+    """
+
+    @abc.abstractmethod
+    def __str__(self):
+        raise NotImplementedError()
+
+    @property
+    def type(self):
+        """A typestring as defined by the apacheconfig parser.
+
+        A single *Node class can have multiple possible values for this. For
+        instance, ItemNode is a generic representation of any single directive
+        but has a couple different possible `type`s, including `comment`,
+        `statement`, and `include`-- depending on which the caller may
+        decide to treat ItemNode differently.
+        """
+        if self._type is None:
+            raise NotImplementedError()
+        return self._type
+
+    @property
+    def whitespace(self):
+        """Trailing or preceding whitespace.
```

Comment:
It is not clear from the docstring how many whitespaces will be reported. If it's just one, how the rest would be represented?